HULKs/hulk
# Recording of robotics nodes

## Introduced Changes

- `PersistentState` is renamed to `CyclerState` because it stores state in a cycler
- Node states, setup main outputs, and cross-inputs (from other cyclers) are recorded by serializing to Bincode and writing it to disk
- `hardware.json` is split into `framework.json` and `hardware.json`
- Recording can be enabled by changing the `framework.json`

## ToDo / Known Issues

- Naming of the structs?
- Fix Pepsi parameter setting

## Ideas for Next Iterations (Not This PR)

- Delete localization recording
- Change pixel storage
- Reuce frequency
- Same seconds in `launchHULK` log files and recording file names
- Reduce recording interleavement
- Add channel directly from robotics domain to "runtime" framework

## How to Test

Deploy to a NAO, check that the cyclers are enabled for recording. When the NAO is in playing primary state, the NAO records to `logs` directory in Bincode format.
